flamingpy/codes/surface_code.py

```diff
@@ -298,9 +304,14 @@ def __init__(
         self.ec = [ec]
         # self.ec = ["primal", "dual"] if ec == "both" else [ec]
 
+        if {ec, boundaries} == {"primal", "dual"}:
```

Comment:
What I am trying to catch here is the condition [mentioned by @soosub](https://github.com/XanaduAI/flamingpy/pull/62#issuecomment-1151480286) on which an uninformative error is raised

> Just a note that the combination of ec='primal' and boundaries='dual' and the converse (ec='dual' and boundaries='primal') result in a KeyError. I discussed this with Ilan and it should not work (so that's good 😃), but maybe it would be good to throw a more interpretable warning.

This suggestion does improve the sanity check. Thanks!